### ISSUE TITLE
Feature request: Add DeepL as alternative translation option for FOSS version

### ISSUE THREAD

**Issue by huuhaa** [Telegram-FOSS-Team/Telegram-FOSS#593]
I would love to see alternative translation option in FOSS version of Telegram, so please add:

 https://www.deepl.com

I also suppose I wouldn't be only one who would like that. It might be totally different story for non FOSS version of Telegram tough, but I assume most FOSS version users would welcome this option.

**Comment by contracourse** [comment]
Yes I agree, unfortunately I also rely on Telegram translations for certain groups and I don't want to use the default Google API. 
I noticed that Forkgram, the other Telegram Foss Fork uses DuckDuckGo as the default translation option, which I'm also in favor of. 
Maybe the Devs can just copy/past their commits into this Foss version. The link is here 
https://github.com/Forkgram/TelegramAndroid/commit/d3f636304deeb11f19f470ed31dba0c6ce91e7e3

**Comment by theoware** [comment]
DeepL also provides much more accurate translations

**Comment by huuhaa** [comment]
There's also Nekogram X which apparently already has DeepL included, maybe that code could be reused?

https://github.com/NekoX-Dev/NekoX

That might be good alternative client too, but I haven't had any issues with Telegramm FOSS, so I haven't switch or even test that one yet. Hoping DeepL could be included to Telegram FOSS.

**Comment by theoware** [comment]
> code could be reused

https://github.com/NekoX-Dev/NekoX/commit/f4a932f06b11db78deef11dbd0b7929f8789c785 Could help

**Comment by contracourse** [comment]
libre translate might also be a good alternative, its fully FOSS and the translation results are not that bad
https://libretranslate.com/
